util/test/activate_chpl_test_venv.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+import os
+import sys
+
+chplenv_dir = os.path.join(os.path.dirname(__file__), '..', 'chplenv')
+sys.path.insert(0, os.path.abspath(chplenv_dir))
+
+import utils
+import chpl_platform
+import chpl_make
+
+# Activate a virtualenv that has testing infrastructure requirements installed
+#
+# A user can set CHPL_TEST_DONT_USE_VENV to skip the virtualenv activation. By
+# doing so they're asserting that all the requirements installed locally.
+#
+# Otherwise we look for a virtualenv to activate. By default, we will try to
+# use $CHPL_HOME/third-party-chpl-venv and check for a sentinel file that
+# test-venv creates when it's been successfully installed. A user can also
+# specify a custom virtualenv. Again, they're asserting that the virtualenv has
+# all the requirements installed.
+#
+# Note that this method does not allow us to specify python versions to use and
+# instead will use the system default. Long term we probably want to have a
+# wrapper script that activates the virtualenv and just calls start_test.
+
+def activate_venv():
+
+    skip_venv_var = 'CHPL_TEST_DONT_USE_VENV'
```

Comment:
Would just keying off `CHPL_TEST_VENV_DIR` being empty or `system` be a little nicer here? I'd to try to toning down the number of env vars we look at.